demo/src/components/demos/layer-base.js

```diff
@@ -0,0 +1,101 @@
+import React, {Component} from 'react';
+import DeckGL from 'deck.gl';
+import autobind from 'autobind-decorator';
+
+import {MAPBOX_STYLES} from '../../constants/defaults';
+import {getLayerParams} from '../../utils/layer-params';
+
+const defaultViewport = {
+  mapStyle: MAPBOX_STYLES.LIGHT,
+  longitude: -122.4,
+  latitude: 37.7,
+  zoom: 9,
+  maxZoom: 20,
+  pitch: 0,
+  bearing: 0
+};
+
+export default function createLayerDemoClass(settings) {
+
+  const renderLayer = (data, params, extraProps = {}) => {
```

Comment:
I'd avoid local classes and functions unless they are very small as I feel they reduce readability.

The absence of the `export` keyword already makes it very clear what is local to a module.